### PR TITLE
feat(igxGridToolbar) Adding the IgxGrid's toolbar feature

### LINKED ISSUE #749: Add Toolbar to the igxGrid
- [ ] Column Pinning UI - #714

### PR DESCRIPTION
**Pull request by rmkrmk**
Closes #749

Adding the IgxGrid's toolbar feature.
- New component: IgxGridToolbarComponent
- New interface: IGridToolbarExportEventArgs
- New inputs added to IgxGridComponent:
  - showToolbar
  - toolbarTitle
  - exportExcel
  - exportCsv
  - exportText
  - exportExcelText
  - exportCsvText
- New outputs added to the IgxGridComponent:
  - onToolbarExporting - EventEmitter<IGridToolbarExportEventArgs>

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grid/grid-toolbar.component.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        return (this.grid != null && (this.grid.exportExcel || this.grid.exportCsv));
+    }
+
+    public get shouldShowExportExcelButton(): boolean {
+        return this.grid.exportExcel;
+    }
+
+    public get shouldShowExportCsvButton(): boolean {
+        return this.grid.exportCsv;
+    }
```
Comment (DDavidkov) [inline projects/igniteui-angular/src/lib/grid/grid-toolbar.component.ts]: All the other fields except shouldShowExportExcelButton and shouldShowExportCsvButton have null checks for the grid. Shouldn't these two also have such check?

File: projects/igniteui-angular/src/lib/grid/grid-toolbar.component.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        this.toggleDirective.collapsed = !this.toggleDirective.collapsed;
+        const args = { grid: this.grid, exporter: this.csvExporter, type: 'csv', cancel: false };
+        this.grid.onToolbarExporting.emit(args);
+        if (args.cancel) {
+            return;
+        }
+        this._exportEventSubscription = this.csvExporter.onExportEnded.subscribe((ev) => this._exportEndedHandler());
+        // show busy indicator here
+        this.csvExporter.export(this.grid, new IgxCsvExporterOptions('ExportedData', CsvFileTypes.CSV));
+    }
```
Comment (DDavidkov) [inline projects/igniteui-angular/src/lib/grid/grid-toolbar.component.ts]: A lot of code duplication in the exportToXXXClicked methods. They can be easily simplified by adding a private method:

``` TypeScript
private export(exporter: IgxBaseExporter, exportType: string) {
        this.exportClicked();
        const args = { grid: this.grid, exporter: exporter, type: exportType, cancel: false };
        this.grid.onToolbarExporting.emit(args);
        if (args.cancel) {
            return;
        }
        this._exportEventSubscription = exporter.onExportEnded.subscribe((ev) => this._exportEndedHandler());
        // show busy indicator here
       const fileName = 'ExportedData';
       const options = exportType === 'excel' ? new IgxExcelExporterOptions(fileName) : new IgxCsvExporterOptions(fileNam , CsvFileTypes.CSV);
       exporter.export(this.grid, );
 }
```
And calling it from the public ones with the appropriate arguments.

File: projects/igniteui-angular/src/lib/grid/grid.component.ts [hunk 3]
```diff
@@ -448,6 +457,97 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
...
+    @ViewChild('toolbar', { read: ElementRef })
+    private toolbarHtml: ElementRef = null;
+
+    @Input()
+    public get showToolbar(): boolean {
+        return this._showToolbar;
+    }
+
+    public set showToolbar(newValue: boolean) {
+        this._showToolbar = newValue;
```
Comment (DDavidkov) [inline projects/igniteui-angular/src/lib/grid/grid.component.ts]: Make sure that the value has actually changed here by adding a this._showToolbar !== newValue check. This applies to all setters below.

Comment (rmkrmk) [reply projects/igniteui-angular/src/lib/grid/grid.component.ts]: Checks added to all toolbar related setters.

**Pull request merged**